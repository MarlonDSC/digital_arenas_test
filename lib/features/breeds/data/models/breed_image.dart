import 'package:json_annotation/json_annotation.dart';

part 'breed_image.g.dart';

@JsonSerializable()
class BreedImage {
  final String id;
  final String url;
  final int width;
  final int height;
  @JsonKey(name: 'mime_type')
  final String? mimeType;

  BreedImage({
    required this.id,
    required this.url,
    required this.width,
    required this.height,
    required this.mimeType,
  });

  factory BreedImage.fromJson(Map<String, dynamic> json) => _$BreedImageFromJson(json);
  Map<String, dynamic> toJson() => _$BreedImageToJson(this);
}