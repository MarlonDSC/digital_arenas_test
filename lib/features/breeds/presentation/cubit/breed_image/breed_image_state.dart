part of 'breed_image_cubit.dart';

sealed class BreedImageState extends Equatable {
  const BreedImageState();

  @override
  List<Object> get props => [];
}

final class BreedImageInitial extends BreedImageState {}

final class BreedImageLoading extends BreedImageState {}

final class BreedImageLoaded extends BreedImageState {
  final BreedImage breedImage;

  const BreedImageLoaded({required this.breedImage});
}

final class BreedImageError extends BreedImageState {
  final List<BaseError> errors;

  const BreedImageError({required this.errors});
}