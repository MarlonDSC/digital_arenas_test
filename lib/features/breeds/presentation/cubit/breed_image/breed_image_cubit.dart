import 'package:bloc/bloc.dart';
import 'package:equatable/equatable.dart';
import 'package:inmo_mobile/core/errors/base_error.dart';
import 'package:inmo_mobile/features/breeds/data/models/breed_image.dart';
import 'package:inmo_mobile/features/breeds/domain/usecases/get_breed_image.dart';

part 'breed_image_state.dart';

class BreedImageCubit extends Cubit<BreedImageState> {
  final GetBreedImageUseCase _getBreedImageUseCase;

  BreedImageCubit(this._getBreedImageUseCase) : super(BreedImageInitial());

  Future<void> getBreedImage(String id) async {
    emit(BreedImageLoading());
    final result = await _getBreedImageUseCase(id);
    result.fold(
      (l) => emit(BreedImageError(errors: l)),
      (r) => emit(BreedImageLoaded(breedImage: r)),
    );
  }
}
