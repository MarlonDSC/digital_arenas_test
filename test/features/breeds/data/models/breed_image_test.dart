import 'package:flutter_test/flutter_test.dart';
import 'package:approval_tests/approval_tests.dart';
import 'package:inmo_mobile/features/breeds/data/models/breed_image.dart';

void main() {
  group('BreedImage', () {
    test('verify basic serialization', () {
      final breedImage = BreedImage(
        id: 'B1iWlZRXQ',
        url: 'https://images.dog.ceo/breeds/retriever-golden/n02099601_1024.jpg',
        width: 1024,
        height: 768,
        mimeType: 'image/jpeg',
      );

      Approvals.verifyAsJson(
        breedImage,
        options: const Options(
          reporter: DiffReporter(),
        ),
      );
    });

    test('verify serialization with null mimeType', () {
      final breedImage = BreedImage(
        id: 'B2iWlZRXQ',
        url: 'https://images.dog.ceo/breeds/retriever-golden/n02099601_1024.jpg',
        width: 800,
        height: 600,
        mimeType: null,
      );

      Approvals.verifyAsJson(
        breedImage,
        options: const Options(
          reporter: DiffReporter(),
        ),
      );
    });

    test('verify JSON roundtrip', () {
      final originalBreedImage = BreedImage(
        id: 'B3iWlZRXQ',
        url: 'https://images.dog.ceo/breeds/retriever-golden/n02099601_1024.jpg',
        width: 1200,
        height: 900,
        mimeType: 'image/png',
      );

      final json = originalBreedImage.toJson();
      final deserializedBreedImage = BreedImage.fromJson(json);

      Approvals.verifyAsJson(
        {
          'original': originalBreedImage,
          'serialized': json,
          'deserialized': deserializedBreedImage,
        },
        options: const Options(
          reporter: DiffReporter(),
        ),
      );
    });

    test('verify fromJson factory with sample payload', () {
      const json = {
        'id': 'B4iWlZRXQ',
        'url': 'https://images.dog.ceo/breeds/retriever-golden/n02099601_1024.jpg',
        'width': 640,
        'height': 480,
        'mime_type': 'image/jpeg',
      };

      final breedImage = BreedImage.fromJson(json);

      Approvals.verifyAsJson(
        {
          'input': json,
          'parsed': breedImage,
        },
        options: const Options(
          reporter: DiffReporter(),
        ),
      );
    });

    test('verify fromJson factory with null mimeType', () {
      const json = {
        'id': 'B5iWlZRXQ',
        'url': 'https://images.dog.ceo/breeds/retriever-golden/n02099601_1024.jpg',
        'width': 1920,
        'height': 1080,
      };

      final breedImage = BreedImage.fromJson(json);

      Approvals.verifyAsJson(
        {
          'input': json,
          'parsed': breedImage,
        },
        options: const Options(
          reporter: DiffReporter(),
        ),
      );
    });
  });
}