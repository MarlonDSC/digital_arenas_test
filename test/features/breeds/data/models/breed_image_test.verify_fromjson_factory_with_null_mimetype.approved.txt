# This file was generated by approval_tests. Please do not edit.

{
  "input": {
    "id": "B5iWlZRXQ",
    "url": "https://images.dog.ceo/breeds/retriever-golden/n02099601_1024.jpg",
    "width": 1920,
    "height": 1080
  },
  "parsed": {
    "id": "B5iWlZRXQ",
    "url": "https://images.dog.ceo/breeds/retriever-golden/n02099601_1024.jpg",
    "width": 1920,
    "height": 1080,
    "mime_type": null
  }
}